release/packaging/utils/publish-debs.sh

```diff
@@ -17,28 +21,50 @@ else
     # -ti to docker-run, and $SECRET_KEY must not require a pass phrase.
     interactive=
 fi
-docker run --rm ${interactive} -v "${rootdir}:/code" -v "${keydir}:/keydir" -w /code/release/packaging/output calico-build/focal /bin/sh -c "gpg --import --batch < /keydir/key && debsign -k'*@' --re-sign *_*_source.changes"
 
 for series in focal jammy noble; do
+    echo "Processing uploads for ${series}"
+    echo
     # Get the packages and versions that already exist in the PPA, so we can avoid
     # uploading the same package and version as already exist.  (As they would be rejected
     # anyway by Launchpad.)
-    sources_url="http://ppa.launchpad.net/project-calico/${REPO_NAME}/ubuntu/dists/${series}/main/source/Sources.gz"
-    existing_packages=$(wget -q -O - ${sources_url} | gzip -d | awk '/^Package:/{printf("%s_", $2);} /^Version:/{sub(/^1:/,"", $2); print $2;}')
-    echo "Existing source packages for ${series} in project-calico/${REPO_NAME} are:"
-    echo "${existing_packages}"
-
-    # Use the Distribution header to map changes files to Ubuntu versions, as some of our
-    # packages don't include the Ubuntu version name in the changes file name.
-    # shellcheck disable=SC2013
-    for changes_file in $(grep -l "Distribution: ${series}" ./*_source.changes); do
-        already_exists=false
-        for existing in ${existing_packages}; do
-            if [ "${changes_file}" = "${existing}_source.changes" ]; then
-                already_exists=true
-                break
-            fi
+    sources_url="https://ppa.launchpadcontent.net/project-calico/${REPO_NAME}/ubuntu/dists/${series}/main/source/Sources.gz"
+    if curl -fs -I "${sources_url}" > /dev/null; then
+        existing_packages=$(curl -fsSL "${sources_url}" | zcat | awk '/^Package:/{printf("%s_", $2);} /^Version:/{sub(/^2:/,"", $2); print $2;}')
+        mapfile -t existing_packages <<< "${existing_packages}"
+        for existing_package in "${existing_packages[@]}"; do
+            # echo "Launchpad has sources for ${existing_package} already, marking to skip" | ts "[check ${series} ${existing_package}]"
+            touch "${outputdir}/${existing_package}_source.ppa.previously-uploaded"
         done
-        ${already_exists} || docker run --rm -v "${rootdir}:/code" -w /code/release/packaging/output calico-build/${series} dput -u "ppa:project-calico/${REPO_NAME}" "${changes_file}" | ts "[upload $series]"
+    fi
+
+    # Loop through our `changes` files (one each for felix and networking-calico, ideally),
```

Comment:
Just FTR: yes, that's true now, but it wasn't in the past when we also had to build dnsmasq, etcd etc. for various reasons; and it is possible it could become untrue again in the future.  E.g. one intermittent ask is that we package up calicoctl.

Reply:
Might be worth deleting or clarifying that parenthesis, in case someone feels inclined to hardcode that it's just felix and networking-calico.